pyaedt/application/DesignSolutions.py

```diff
@@ -0,0 +1,678 @@
+
+    @aedt_exception_handler
+    def set_auto_open(self, enable=True, boundary_type="Radiation"):
+        options = ["NAME:Options", "EnableAutoOpen:=", enable]
+        if enable:
+            options.append("BoundaryType:=")
+            options.append(boundary_type)
+        self._solution_options[self._solution_type]["options"] = options
+        self.solution_type = self._solution_type
+
+
+class Maxwell2DDesignSolution(DesignSolution, object):
+    def __init__(self, odesign, design_type, aedt_version):
+        DesignSolution.__init__(self, odesign, design_type, aedt_version)
+        self._geometry_mode = "XY"
+
+    @property
+    def xy_plane(self):
+        return self._geometry_mode == "XY"
+
+    @xy_plane.setter
+    @aedt_exception_handler
+    def xy_plane(self, val=True):
+        if val:
+            self._geometry_mode = "XY"
+        else:
+            self._geometry_mode = "about Z"
+        self._solution_options[self._solution_type]["options"] = self._geometry_mode
+        self.solution_type = self._solution_type
+
+    @property
+    def solution_type(self):
+        return self._solution_type
+
+    @solution_type.setter
+    @aedt_exception_handler
+    def solution_type(self, soltype):
+        if soltype is None:
+            if "GetSolutionType" in dir(self._odesign):
+                self._solution_type = self._odesign.GetSolutionType()
+                if "Modal" in self._solution_type:
+                    self._solution_type = "Modal"
+                elif "Terminal" in self._solution_type:
+                    self._solution_type = "Terminal"
+            return
+        if soltype[-1:] == "Z":
+            self._solution_type = soltype[:-1]
+            self._solution_options[self._solution_type]["options"] = "about Z"
+            self._geometry_mode = "about Z"
+        elif soltype[-2:] == "XY":
+            self._solution_type = soltype[:-2]
+            self._solution_options[self._solution_type]["options"] = "XY"
+            self._geometry_mode = "XY"
+        if self._solution_type in self._solution_options and self._solution_options[self._solution_type]["name"]:
+            try:
+                opts = (
+                    ""
+                    if self._solution_options[soltype]["options"] is None
+                    else self._solution_options[soltype]["options"]
+                )
+                self._odesign.SetSolutionType(self._solution_options[soltype]["name"], opts)
+            except:
+                pass
+
+
+class IcepakDesignSolution(DesignSolution, object):
+    def __init__(self, odesign, design_type, aedt_version):
+        DesignSolution.__init__(self, odesign, design_type, aedt_version)
+        self._problem_type = "TemperatureAndFlow"
+
+    @property
+    def problem_type(self):
+        return self._problem_type
+
+    @problem_type.setter
+    @aedt_exception_handler
+    def problem_type(self, val="TemperatureAndFlow"):
```

Comment:
Most of these of these methods are missing examples. Please have them documented either in his PR or in a follow up. Pinging @PipKat for help.

Reply:
@akaszynski and @maxcapodi78 
There has been a lot of refactoring and I was planning on eventually going through all the AEDT doc to look for inconsistencies. However, I can try to quickly look at and edit docstrings in this PR.

Reply:
I was still fixing the issues due to the big refactoring. I'm adding docstrings now